src/inmanta/parser/plyInmantaLex.py

```diff
@@ -162,7 +181,19 @@ def t_INT(t: lex.LexToken) -> lex.LexToken:  # noqa: N802
 
 def t_STRING(t: lex.LexToken) -> lex.LexToken:  # noqa: N802
     r"(\"([^\\\"\n]|\\.)*\")|(\'([^\\\'\n]|\\.)*\')"
-    t.value = bytes(t.value[1:-1], "utf-8").decode("unicode_escape")
+    try:
+        with warnings.catch_warnings():
+            warnings.filterwarnings("error", message="invalid escape sequence")
+            value: str = bytes(t.value[1:-1], "utf-8").decode("unicode_escape")
+    except DeprecationWarning:
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore")
+            value = bytes(typing.cast(str, t.value)[1:-1], "utf_8").decode("unicode_escape")
+
+        msg: str = f"({t.lexer.inmfile}:{t.lexer.lineno}) Invalid escape sequence in string:\n{value}"
+        warnings.showwarning(msg, lineno=t.lexer.lineno, filename=t.lexer.inmfile, category=ParserWarning)
```

Comment:
Should I factor this and the MLS code into a function ? Would that add a performance overhead ?

Reply:
If you can make it a clean function I would indeed go for that. I don't think the slight performance overhead should be a concern during parsing. @wouterdb cc.

Reply:
1. The function call I  would not worry about.
2. The context manager, I would check what it does to parser performance.

Reply:
Here are the results of the performance benchmark I did:
Model |Master | Feature Branch |
|-----|-----|---|
| Small model with 3 bad escapes |  54.8910 ms  | 57.0080 ms 
| Bigger model with 12 bad escapes |  56.7250 ms  | 59.9952 ms 
| Bigger model with no bad escapes |  54.7620 ms  | 55.1883 ms 

I will paste the full results on the main thread.

Reply:
@wouterdb wdyt ?